src/Box_TEST.cc

```diff
@@ -145,6 +145,51 @@ TEST(DOMBox, Shape)
   EXPECT_EQ(gz::math::Vector3d(1, 2, 3), box.Size());
 }
 
+/////////////////////////////////////////////////
+TEST(DOMBox, CalculateInertial)
+{
+  sdf::Box box;
+
+  // density of aluminium
+  double density = 2710;
+
+  // Invalid dimensions leading to std::nullopt return
+  // CalculateInertials()
+  box.SetSize(gz::math::Vector3d(-1, 1, 0));
+  auto invalidBoxInertial = box.CalculateInertial(density);
+  ASSERT_EQ(std::nullopt, invalidBoxInertial);
+
+  double l = 2;
+  double w = 2;
+  double h = 2;
+  box.SetSize(gz::math::Vector3d(l, w, h));
+
+  double expectedMass = box.Shape().Volume() * density;
+  double ixx = (1.0/12.0) * expectedMass * (w*w + h*h);
+  double iyy = (1.0/12.0) * expectedMass * (l*l + h*h);
+  double izz = (1.0/12.0) * expectedMass * (l*l + w*w);
+
+  gz::math::MassMatrix3d expectedMassMat(
```

Comment:
include `MassMatrix3d`,gz::math::Vector3d and gz::math::Inertiald and gz::math::Pose3d, optional